src/rpc/server.cpp

```diff
@@ -456,6 +456,163 @@ UniValue CRPCTable::execute(const JSONRPCRequest &request) const
+            "getblockfilter",
+            "getblockhash",
+            "getblockheader",
+            "getblockstats",
+            "getblocktemplate",
+            "getchaintips",
+            "getchaintxstats",
+            "getconnectioncount",
+            "getdescriptorinfo",
+            "getdifficulty",
+            "getindexinfo",
+            "getmemoryinfo",
+            "getmempoolancestors",
+            "getmempooldescendants",
+            "getmempoolentry",
+            "getmempoolinfo",
+            "getmininginfo",
+            "getnettotals",
+            "getnetworkhashps",
+            "getnetworkinfo",
+            "getnewaddress",
+            "getnodeaddresses",
+            "getpeerinfo",
+            "getrawchangeaddress",
+            "getrawmempool",
+            "getrawtransaction",
+            "getreceivedbyaddress",
+            "getreceivedbylabel",
+            "getrpcinfo",
+            "gettransaction",
+            "gettxout",
+            "gettxoutproof",
+            "gettxoutsetinfo",
+            "getunconfirmedbalance",
+            "getwalletinfo",
+            "getzmqnotifications",
+            "help",
+            "importaddress",
+            "importdescriptors",
+            "importprunedfunds",
+            "importpubkey",
+            "importwallet",
+            "invalidateblock",
+            "joinpsbts",
+            "keypoolrefill",
+            "listaddressgroupings",
+            "listbanned",
+            "listlabels",
+            "listlockunspent",
+            "listreceivedbyaddress",
+            "listreceivedbylabel",
+            "listsinceblock",
+            "listtransactions",
+            "listunspent",
+            "listwalletdir",
+            "listwallets",
+            "loadwallet",
+            "lockunspent",
+            "logging",
+            "mockscheduler",
+            "ping",
+            "preciousblock",
+            "prioritisetransaction",
+            "pruneblockchain",
+            "psbtbumpfee",
+            "reconsiderblock",
+            "removeprunedfunds",
+            "rescanblockchain",
+            "savemempool",
+            "scantxoutset",
+            "sendmany",
+            "sendrawtransaction",
+            "sendtoaddress",
+            "setban",
+            "setlabel",
+            "setmocktime",
+            "setnetworkactive",
+            "settxfee",
+            "setwalletflag",
+            "stop",
+            "submitblock",
+            "submitheader",
+            "testmempoolaccept",
+            "unloadwallet",
+            "upgradewallet",
+            "uptime",
+            "utxoupdatepsbt",
+            "validateaddress",
+            "verifychain",
+            "verifymessage",
+            "verifytxoutproof",
+            "waitforblock",
+            "waitforblockheight",
+            "waitfornewblock",
+            "walletcreatefundedpsbt",
+            "walletlock",
+            "walletprocesspsbt",
+        };
```

Comment:
This allow list came from listing all the RPCs and then removing the following:
```
signmessagewithprivkey
signrawtransactionwithkey
signrawtransactionwithwallet
signmessage
createwallet
encryptwallet
importmulti
importprivkey
sethdseed
walletpassphrase
walletpassphrasechange
```
As far as I can tell, these are the RPCs whose arguments shouldn't be logged.

Reply:
Maintainence of this list seems like a drawback. It would be nice if a test fails somewhere when an RPC is added or removed without the allow list being updated, e.g. with a list of structs of all the RPCs with an associated safe/unsafe bool value, or something like `CRPCConvertParam`. Feel to ignore if unrealistic.